### ISSUE TITLE
Slow performance on FlatfileClient.records.get method

### ISSUE THREAD

**Issue by thyen** [FlatFilers/flatfile-node#44]
Hello everyone,

Currently we are in the process of upgrading our flatfile usage, and in doing so we've found performance issues while using the `records.get` method to fetch the records from a sheet.

For reference we're using the following file, which contains anonymized data:
[import-template (6).csv](https://github.com/FlatFilers/flatfile-node/files/14888768/import-template.6.csv)

When running `FlatfileClient.records.get(sheetId, { pageSize: 10000, pageNumber: 1, includeMessages: false })` , the method takes around 50 to 60 seconds to execute. Debugging the code we identifier that the API request (execute [here](https://github.com/FlatFilers/flatfile-node/blob/1.7.5/src/api/resources/records/client/Client.ts#L143-L162)) takes around 3 to 4 seconds to execute, which match our expectations. The remainder of the execution time seems to be spent on the serialization of the response (executed on [parseOrThrow](https://github.com/FlatFilers/flatfile-node/blob/1.7.5/src/core/schemas/builders/schema-utils/getSchemaUtils.ts#L21-L29) and subsequently on [validateAndTransformRecord](https://github.com/FlatFilers/flatfile-node/blob/1.7.5/src/core/schemas/builders/record/record.ts#L84-L137)).

Not only the performance is slow, but when running the request on our micro-service in Kubernetes the pod is crashing with out-of-memory errors, said pod is currently provisioned with 500 MB of RAM which was previously enough to run the requests. Although we did not profile the memory usage, when we fallback to making the API request directly and returning the response of the API request, our code works as expected with the provisioned resources.

In order to check the performance we added a few `console.time` logs on these methods (see [this commit](https://github.com/thyen/flatfile-node/commit/b1685d0df82b00aecd41094256093eab8d724f37)) , created a temporary file with the following code and ran it using `ts-node`:

```
import { FlatfileClient } from "./src/index";

const token = process.env.FLATFILE_TOKEN;
const apiUrl = "https://platform.eu.flatfile.com/api/v1";
const sheetId = process.env.FLATFILE_SHEETID;

async function main() {
    try {
        const client = new FlatfileClient({ token, apiUrl });

        await client.records.get(sheetId, {
            pageSize: 10000,
            pageNumber: 1,
            includeMessages: false,
        });
    } catch (e) {
        console.error(e);
    }

    return 0;
}

main().then(process.exit);

```

We wanted to check with you guys if this is the expected behavior of the library or if there is something else we need to setup in order to get a better performance in our code. As I mentioned, we're currently using the API directly with the `axios` library in order to make the request and return the body of the response, since in our use-case the object returned by the API is good enough to be used.

**Comment by dboskovic** [comment]
Yeah, we're working hard to get the serde layer removed from this endpoint. For large record fetches we are currently bypassing this ourselves in our own implementations for the same reasons.

**Comment by dsinghvi** [comment]
Hey @thyen -- we're working on a patch to update the validation and will run it through the sample data you linked. Thank you for filing such a detailed issue 🙏
